packages/table/src/table.tsx

```diff
@@ -338,6 +336,50 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         selectionModes: SelectionModes.ALL,
     };
 
+    private static SHALLOWLY_COMPARABLE_PROP_KEYS = [
```

Comment:
@giladgray @gscshoyru I have @giladgray's suggestion working, but it requires changes all over the codebase. Can I push them up in a separate PR? Can open immediately after this one merges.